R/module_nested_tabs.R

```diff
@@ -368,3 +373,41 @@ srv_teal_module.teal_module <- function(id,
     )
   }
 }
+
+#' Calls expression when condition is met
+#'
+#' Function postpones `handlerExpr` to the moment when `eventExpr` (condition) returns `TRUE`,
+#' otherwise nothing happens.
+#' @param condExpr logical expression determining moment when `handlerExpr` should be evaluated.
+#' @param ... additional arguments passed to `observeEvent` with the exception of `eventExpr` that is not allowed.
+#' @inheritParams shiny::observeEvent
+#'
+#' @return An observer.
+#'
+#' @keywords internal
+call_once_when <- function(condExpr,
+                           handlerExpr,
+                           event.env = parent.frame(),
+                           handler.env = parent.frame(),
+                           ...) {
+  if ("eventExpr" %in% names(rlang::list2(...))) {
+    stop("eventExpr is not allowed in call_once_when.")
+  }
+
+  event_quo <- rlang::new_quosure(substitute(condExpr), env = event.env)
```

Comment:
```suggestion
  event_quo <- rlang::new_quosure(substitute(condExpr), env = parent.frame(2))
```

Reply:
It's twofold:

1. Allows for full compatibility with `observeEvent` it calls
    - The environments are not passed on to `observeEvent` via `...`
    - In case a custom environment is needed (not currently) this ensures that quosures include correct environment
1. I believe it's good practice to have this on the arguments (see packages `checkmate`, `shiny`)

Reply:
Fair enough 👍

Reply:
Shouldn't `event.env` and `handler.env` be passed to `observeEvent` then instead of `quosure`?

Reply:
fair enough 👍